File: apps/capability-host/scripts/operational-environment.mjs
const LEGACY_PREFIX = 'HULL_'
const CAPABILITY_HOST_PREFIX = 'CAPABILITY_HOST_'

/** Refuse stale capability-host configuration instead of silently ignoring it. */
export function assertNoLegacyOperationalVariables(env = process.env) {
  const legacyName = Object.keys(env)
    .filter((name) => name.startsWith(LEGACY_PREFIX))
    .sort()[0]
  if (legacyName == null) return

  const replacement = CAPABILITY_HOST_PREFIX + legacyName.slice(LEGACY_PREFIX.length)
  throw new Error(
    `Legacy capability-host environment variable ${legacyName} is not supported; use ${replacement}.`,
  )
}

File: .claude/worktrees/land-58214-1788861322/apps/capability-host/src/runtime/operational_environment.py
"""Clean-break guard shared by every standalone capability-host Python worker."""

import os

_LEGACY_PREFIX = "HULL_"
_REPLACEMENT_PREFIX = "CAPABILITY_HOST_"


def assert_no_legacy_operational_variables(environment=None):
    """Refuse stale capability-host configuration before any worker reads its env."""
    environment = os.environ if environment is None else environment
    legacy_name = next(
        (name for name in sorted(environment) if name.startswith(_LEGACY_PREFIX)),
        None,
    )
    if legacy_name is None:
        return
    replacement = _REPLACEMENT_PREFIX + legacy_name[len(_LEGACY_PREFIX):]
    raise RuntimeError(
        f"Legacy capability-host environment variable {legacy_name} is not supported; "
        f"use {replacement}."
    )
